### DISCUSSION TITLE
steam: gamepad works in the steam client but not in game

### DISCUSSION THREAD

**Discussion by opqriu** [netblue30/firejail#6523] [Q&A] [answered]
Hello, I am running Firejail Steam!

It works by default, but...

Using "Steam Input".... For some games... 

the gamepad works in the Steam Client, but it doesn’t work in the game.

Is there a way to completely allow the gamepad to function in Firejail?

```sh
# I’m not sure if this will help with the answer, but below are related to the
# gamepad:
nano /etc/udev/rules.d/99-8bitdo-xinput.rules
udevadm control --reload-rules && sudo udevadm trigger
```

Relates to:

* #2448
* #6534

**Comment by kmk3** [comment, marked as answer]
Basic debugging information is missing; please follow the bug report template:

* <https://github.com/netblue30/firejail/issues/new?template=bug_report.md>

> the gamepad works in the Steam Client, but it doesn’t work in the game.

Which game?

Does it work if you add the following to ~/.config/firejail/steam.local?

```
ignore nou2f
```

If not, does it work with the following?

```
ignore nou2f
ignore private-dev
```

Edit: Should be fixed by:

* #6534

**Reply by opqriu** [reply]
~/.config/firejail/steam.local?
> 
> ```
> ignore nou2f
> ```


The relevant games are "**Undertale**", "**Persona 4 Golden**", and "**Persona 5 Royal**". 

Also, the "**ignore nou2f**" option worked successfully. (~/.config/firejail/steam.local)

Now, when I launch these games through Firejail Steam, the gamepad works immediately. 

Thank you!

**Comment by kmk3** [comment]
(Offtopic)

Please see the following links for how to format code blocks in markdown:

* <https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks>
* <https://github.github.com/gfm/#fenced-code-blocks>
